### DISCUSSION TITLE
Feature request - Database engines

### DISCUSSION THREAD

**Discussion by tinola** [sinamics/ztnet#163] [Ideas]
As we know, so far `ztnet` supports `PostgreSQL` database only.
It'll be nice to have possibility to use other engines like `MySQL/MariaDB`, even `SQLite` for small ZT networks - especially using standalone version of `ztnet` for users who have those database engines already installed.
Perhaps someone of the advanced users would be able to contribute or prepare appropriate startup configurations?....

**Comment by sinamics** [comment]
The primary challenge lies in the limited support for JSON data types in these databases compared to PostgreSQL:

- MySQL: Has JSON support from version 5.7.8, but there might be some limitation versus Postgres.
- SQLite: It does not have a native JSON data type and only offers limited JSON functions.
- MariaDB: Its JSON support is quite limited, storing JSON data as LONGTEXT.

Given these limitations, adding support for these databases while maintaining the current functionalities of ztnet would be a complex task.

https://github.com/sinamics/ztnet/blob/main/prisma/schema.prisma

**Comment by tinola** [comment]
Yeah... so, the appeal! **Let's make `ztnet` better and better!** 👍
Who will take up the challenge with adding MySQL/MariaDB support?... Come on community, ztnet is worth it 😃
